### PR TITLE
provide rustfmt.toml

### PR DESCRIPTION
**Pull request by maxcountryman**
For better style consistency, this introduces a rustfmt.toml.

### CODE REVIEW

**Comment by maxcountryman** [comment]
@LeoniePhiline I'd like to introduce a `rustfmt.toml` to encourage consistency with imports and comment line wrapping. If this is too noisy, that's okay, it's not super important.

**Comment by LeoniePhiline** [comment]
I think these are a good set, matching my own preferences.

I would like to also propose a rather pedantic linting configuration, as detailed here: https://github.com/zslayton/cron/pull/117#issuecomment-2444128976

File: src/schedule.rs [hunk 1]
```diff
@@ -848,8 +856,8 @@ mod test {
         //let dt = schedule_tz
         //    .ymd(2019, 10, 27)
         //    .and_hms(0, 3, 29)
-        //    .checked_add_signed(chrono::Duration::hours(1)) // puts it in the middle of the DST transition
-        //    .unwrap();
+        //    .checked_add_signed(chrono::Duration::hours(1)) // puts it in the middle
+        // of the DST transition    .unwrap();
```
Comment (LeoniePhiline) [inline src/schedule.rs]: This is an unfortunate wrap that might benefit from a manual adjustment.

In any case, these commented-out upstream lines are going to be removed, so we might as well not bother in this case.

**Comment by LeoniePhiline** [comment]
Does `imports_granularity` also summarize `self::` and `super::` under `crate::`?

I am fine with keeping `self::` and `super::` as context-relative imports, which make sense also semantically.

File: src/parsing.rs [hunk 2]
```diff
@@ -1,28 +1,34 @@
...
 
 impl TryFrom<Cow<'_, str>> for Schedule {
     type Error = Error;
 
     fn try_from(expression: Cow<'_, str>) -> Result<Self, Self::Error> {
         match schedule(&expression) {
-            Ok((_, schedule_fields)) => Ok(Schedule::new(expression.into_owned(), schedule_fields)), // Extract from nom tuple
-            Err(_) => Err(ErrorKind::Expression("Invalid cron expression.".to_owned()).into()), //TODO: Details
+            Ok((_, schedule_fields)) => Ok(Schedule::new(expression.into_owned(), schedule_fields)), /* Extract from nom tuple */
+            Err(_) => Err(ErrorKind::Expression("Invalid cron expression.".to_owned()).into()), /* TODO: Details */
```
Comment (LeoniePhiline) [inline src/parsing.rs:31]: Is this comment-style in any way better? Which setting caused the change?

File: src/schedule.rs [hunk 3]
```diff
@@ -295,37 +296,44 @@ impl Schedule {
             && self.fields.seconds.includes(date_time.second() as Ordinal)
     }
 
-    /// Returns a [TimeUnitSpec] describing the years included in this [Schedule].
+    /// Returns a [TimeUnitSpec] describing the years included in this
+    /// [Schedule].
```
Comment (LeoniePhiline) [inline src/schedule.rs]: We might want to also wrap these (across the entire crate) in backticks. I.e.:

```rust
    /// Returns a [`TimeUnitSpec`] describing the years included in this
    /// [`Schedule`].
```

File: src/time_unit/mod.rs [hunk 4]
```diff
@@ -147,22 +151,24 @@ pub trait TimeUnitSpec {
...
     /// let schedule = Schedule::from_str(expression).expect("Failed to parse expression.");
     ///
     /// assert_eq!(2, schedule.days_of_month().count());
     /// ```
     fn count(&self) -> u32;
 
-    /// Checks if this TimeUnitSpec is defined as all possibilities (thus created with a '*', '?' or in the case of weekdays '1-7')
-    /// # Example
+    /// Checks if this TimeUnitSpec is defined as all possibilities (thus
+    /// created with a '*', '?' or in the case of weekdays '1-7') # Example
```
Comment (LeoniePhiline) [inline src/time_unit/mod.rs]:
```suggestion
    /// created with a '*', '?' or in the case of weekdays '1-7')
    ///
    /// # Example
```

File: src/time_unit/mod.rs [hunk 5]
```diff
@@ -125,12 +126,15 @@ pub trait TimeUnitSpec {
     /// ```
     fn iter(&self) -> OrdinalIter<'_>;
 
-    /// Provides an iterator which will return each included ordinal within the specified range.
-    /// # Example
+    /// Provides an iterator which will return each included ordinal within the
+    /// specified range. # Example
```
Comment (LeoniePhiline) [inline src/time_unit/mod.rs]:
```suggestion
    /// specified range.
    ///
    /// # Example
```

File: src/time_unit/mod.rs [hunk 6]
```diff
@@ -56,12 +52,15 @@ impl<'a> DoubleEndedIterator for OrdinalRangeIter<'a> {
     }
 }
 
-/// Methods exposing a schedule's configured ordinals for each individual unit of time.
-/// # Example
+/// Methods exposing a schedule's configured ordinals for each individual unit
+/// of time. # Example
```
Comment (LeoniePhiline) [inline src/time_unit/mod.rs]:
```suggestion
/// of time.
///
/// # Example
```

**Comment by LeoniePhiline** [comment]
I am done reviewing.

**Comment by LeoniePhiline** [comment]
> encourage consistency

I'd like to go one step further and enforce it in CI:

- **`rustfmt` check:** `cargo --locked fmt --check`
- **`clippy` check:** `cargo --locked clippy --all-features --all-targets -- --deny warnings`

File: src/parsing.rs (hunk 2 above)
Comment (maxcountryman) [reply src/parsing.rs:31]: Strange. I've never seen this configuration change the comment style before, so I'm not sure why it's doing that. I don't think the change is better.

**Comment by maxcountryman** [comment]
> Does `imports_granularity` also summarize `self::` and `super::` under `crate::`?

It does not. There may be some way to configure that, but I haven't looked into that.

**Comment by LeoniePhiline** [comment]
Do we benefit from `+nightly`?

**Comment by LeoniePhiline** [comment]
Ah, I see.

```
Warning: can't set `wrap_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `format_code_in_doc_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `format_strings = true`, unstable features are only available in nightly channel.
Warning: can't set `imports_granularity = Crate`, unstable features are only available in nightly channel.
Warning: can't set `group_imports = StdExternalCrate`, unstable features are only available in nightly channel.
Warning: can't set `wrap_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `format_code_in_doc_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `format_strings = true`, unstable features are only available in nightly channel.
Warning: can't set `imports_granularity = Crate`, unstable features are only available in nightly channel.
Warning: can't set `group_imports = StdExternalCrate`, unstable features are only available in nightly channel.
```

Alright. :D

**Comment by maxcountryman** [comment]
> Do we benefit from `+nightly`?

Unfortunately, yes:

```
cargo +stable fmt --all -- --check
Warning: can't set `wrap_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `format_code_in_doc_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `format_strings = true`, unstable features are only available in nightly channel.
Warning: can't set `imports_granularity = Crate`, unstable features are only available in nightly channel.
Warning: can't set `group_imports = StdExternalCrate`, unstable features are only available in nightly channel.
Warning: can't set `wrap_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `format_code_in_doc_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `format_strings = true`, unstable features are only available in nightly channel.
Warning: can't set `imports_granularity = Crate`, unstable features are only available in nightly channel.
Warning: can't set `group_imports = StdExternalCrate`, unstable features are only available in nightly channel.
```

**Pull request merged**